docs/REST API/rest-api.md

```diff
@@ -8548,8 +8571,67 @@ Returns information about the specified software version.
 }
 ```
 
-## Vulnerabilities
+### Get available App Store apps
+
+Returns the list of App Store (VPP) apps purchased in Apple Business Manager.
+
+`GET /api/v1/fleet/software/app_store/:team_id`
+
+#### Parameters
+
+| Name    | Type | In | Description |
+| ------- | ---- | -- | ----------- |
+| team_id | integer | query | **Required**. The team ID. Lists available VPP software for specified team. |
+
+#### Example
+
+`GET /api/v1/fleet/software/app_store/3`
+
+##### Default response
 
+`Status: 200`
+
+```json
+{
```

Comment:
Needs pagination meta and default page size 50 I think.

Reply:
We discussed this with @rachaelshaw and decided to don't add pagination to this one. We're pretty sure it's a small list of apps. So far we didn't see more than 10.

Reply:
The time to develop with pagination vs the time to develop without pagination is not even an hour's difference. All list endpoints should start with pagination and if you just want it all bump the `per_page` to 1000. That puts the choice to make those decisions into the api user / end user.

As an engineering organization it is irresponsible of us to make list endpoints without pagination no matter how small.

Reply:
To expand on wireframe-first: we decided to not have pagination in the UI so we don't need it in the API.

We could be wrong and end up adding it later. If we need to add it quickly we can file a `P2` story.

Reply:
While the urge to move fast is understandable, skipping pagination is a short-sighted approach that can lead to significant challenges and costs down the line. The minimal initial effort required to implement pagination is far outweighed by the long-term benefits of scalability, performance, user satisfaction, and cost efficiency. By incorporating pagination from the start, we future-proof the API, avoid technical debt, and position Fleet for sustainable growth and success.

I defer to the Release DRI @lukeheath but I want everyone to be aware that I will raise this every time we receive a list api (no matter how small) that does not include pagination by default.

Reply:
Thanks George. It's not a speed thing. Wireframe/UI-first the way we approach UI/API design at Fleet.

I think it's up to @rachaelshaw (API design DRI).

Reply:
Yeah worth keeping in mind that anything we paginate by convention has the "< Previous" + "Next >" buttons in the UI; if those are always expected to be disabled, or feel out-of-place, then probably doesn't need pagination.

(Also, can always add later if needed.)

Reply:
we can still build a UI that calls /endpoint&page_size=1000 and not develop UI pagination atm but in the backend it should always have pagination to start